packages/teleport/src/lib/tdp/client.ts

```diff
@@ -39,11 +39,10 @@ export enum TdpClientEvent {
 // calling Client.nuke() (typically after Client emits a TdpClientEvent.DISCONNECT or TdpClientEvent.ERROR event) in order to clean
 // up its websocket listeners.
 export default class Client extends EventEmitterWebAuthnSender {
-  codec: Codec;
-  socket: WebSocket;
-  socketAddr: string;
-  username: string;
-  logger = Logger.create('TDPClient');
+  protected codec: Codec;
+  protected socket?: WebSocket;
```

Comment:
what does the question mark do here?

Reply:
I had assumed it meant optional (like in interfaces or types), equivalent to:

```typescript
protected socket: WebSocket | undefined;
```

However it turns out that in order for it to be interpreted that way, we need to add `strictNullChecks` to our `tsconfig.json`, which then causes a ton of other errors elsewhere in the codebase. I think its a good option to turn on, because it prevents us from using any property on a `null` or `undefined`, so I've added an issue for it: https://github.com/gravitational/webapps/issues/957.

I've made it explicit for now, like in the example above.